src/diff/index.js

```diff
@@ -128,7 +128,10 @@ export function diff(parentDom, newVNode, oldVNode, context, isSvg, excessDomChi
 
 			c.base = newVNode._dom;
 
-			while (tmp=c._renderCallbacks.pop()) tmp.call(c);
+			while (tmp=c._renderCallbacks.pop()) {
+				if (c._nextState) { c.state = c._nextState; }
```

Comment:
This is clever. @JoviDeCroock I think I assumed the fix here would exist in setState() - what's the reason for moving it into the update loop?

Reply:
Well we could do it in setState but I thought it would be easier to make this happen in the callbacks personally.

It looked like the correct place for me, also felt like the other was increasing size more drastically. Would you prefer me making the second as an example as well?

Reply:
I guess I would be curious to see the difference. Did you have the other version lying around?

I wonder... we have this at the beginning of setState:

```js
let s = (this._nextState!==this.state && this._nextState) || (this._nextState = assign({}, this.state));
```

Maybe that just needs to be moved into forceUpdate?

Reply:
I can take some time to reproduce it this afternoon, hmm why move that to forceUpdate? The callbacks are fired subsequently no?

Reply:
yeah, just I'm wondering if there's a way to have setState() and the render loop both use the same shared logic, rather than different cases for each. (I might be way off here!)

Reply:
I'll try to see if this is possible, this could maybe be a first step to solving https://github.com/preactjs/preact/issues/1736

Reply:
I think doing so would make `componentDidUpdate` reflect invalid prevState